### ISSUE TITLE
[Documentation] More ways to deploy poktrolld

### ISSUE THREAD

**Issue by okdas** [pokt-network/poktroll#526]
## Objective

Document how to deploy and use poktrolld in different environments.

## Origin Document

Currently we only provide an example of how to deploy in [poktroll-docker-compose-example](https://github.com/pokt-network/poktroll-docker-compose-example) repo, but we also need to add documentation how to run in other environments.

## Goals

- Document how to run fullnode/supplier/appgate without docker (systemd?)
- Document helm charts (v1 helm charts are a good example - they were documented) and provide a simple example on how to push to k8s cluster
- Document how to maintain the validator process with [cosmovisor](https://docs.cosmos.network/main/build/tooling/cosmovisor).

## General deliverables

- [ ] **Comments**: Add/update TODOs and comments alongside the source code so it is easier to follow.
- [ ] **Testing**: Add new tests (unit and/or E2E) to the test suite.
- [ ] **Makefile**: Add new targets to the Makefile to make the new functionality easier to use.
- [ ] **Documentation**: Update architectural or development READMEs; use [mermaid](https://mermaid-js.github.io/mermaid/) diagrams where appropriate.

---

**Creator**: @okdas

**Comment by okdas** [comment]
cc @Olshansk

**Comment by okdas** [comment]
@Olshansk we can separate k8s helm chart docs into the separate ticket

**Comment by Olshansk** [comment]
> @Olshansk we can separate k8s helm chart docs into the separate ticket

Let's do it and mark them as deliverables in that ticket.

**Comment by okdas** [comment]
We must cover https://docs.cosmos.network/main/build/tooling/cosmovisor as this is the sidecar process that is used with `x/upgrade` to upgrade the binary on particular heights.

**Comment by okdas** [comment]
Moving this to the next iteration because I recently realized I want to also add Cosmovisor to the docker-compose setup. It should be quick and easy, but will only become usable after the next release. 

After that — I'll create a separate issue to deploy off-chain actors on fly.io as this is a great platform for our use case.

**Comment by Olshansk** [comment]
@okdas Curious why `fly.io` over `render.com`?

Only asking because I know the founding team in the latter.

**Comment by okdas** [comment]
@Olshansk fly.io offers an entirely different approach to how the service is deployed across regions.

TLDR is their public network IP addresses are [anycast](https://fly.io/docs/networking/services/#anycast-ip-addresses) - basically what GCP MCI offers but natively (meaning not a half-baked solution, but something that `fly.io` is built on from the beginning).

From what I understand, render.com is basically a PaaS.

**Comment by Olshansk** [comment]
@okdas This is really great to know!

I think fly.io might be a good `one click deployment` partner in the future then :)

**Comment by Olshansk** [comment]
Too general. When needed, it'll come back up.
